sdk/core/azure-core/src/http/transport_policy.cpp

```diff
@@ -11,33 +12,157 @@
 #include "azure/core/http/win_http_transport.hpp"
 #endif
 
+#include <sstream>
+#include <string>
+
 using Azure::Core::Context;
 using namespace Azure::Core::IO;
 using namespace Azure::Core::Http;
 using namespace Azure::Core::Http::Policies;
 using namespace Azure::Core::Http::Policies::_internal;
 
-std::shared_ptr<HttpTransport> Azure::Core::Http::Policies::_detail::GetTransportAdapter()
-{
-  // The order of these checks is important so that WinHTTP is picked over libcurl on Windows, when
-  // both are defined.
+namespace Azure { namespace Core { namespace Http { namespace Policies { namespace _detail {
+  namespace {
+    bool AnyTransportOptionsSpecified(TransportOptions const& transportOptions)
```

Comment:
Does Any following our naming guidelines (which are in need of being recorded :))

Nit: AreTransportOptionsSpecified

Reply:
or, `IsAnyTransportOptionSpecified()`?